cxx4opencl/intro.txt

```diff
@@ -1,46 +1,63 @@
 3.2` and `section 3.3` detailing fundamental differences of OpenCL execution
 and memory models from the conventional C and {cpp} view.
 
 This document describes the programming language in details. It is not
 structured as a standalone document, but rather as an addition to OpenCL C
-v2.0 defined in <<openclc-spec, The OpenCL C Specification, Version 2.0>>
-and {cpp}17 defined in <<cpp17-spec, The C++17 Specification>>. Where necessary
-this document refers to the specifications of those languages accordingly.
-A full understanding of {cpp} for OpenCL requires familiarity with the
-specifications or other documentation of both languages that {cpp} for OpenCL
+3.0 unified specification defined in <<openclc-spec, The OpenCL C Specification,
+Version 3.0>> and {cpp}17 defined in <<cpp17-spec, The C++17 Specification>>.
+Where necessary this document refers to the specifications of those languages
+accordingly. A full understanding of {cpp} for OpenCL requires familiarity with
+the specifications or other documentation of both languages that {cpp} for OpenCL
 is built upon.
 
 The description of {cpp} for OpenCL starts from highlighting <<diff2openclc,
 _the differences to OpenCL C_>> and <<diff2cxx, _the differences to {cpp}_>>.
 
 The majority of content covers the behavior that is not documented in
-the OpenCL C v2.0 `section 6` and {cpp}17 specifications. This is mainly
+the OpenCL C 3.0 `section 6` and {cpp}17 specifications. This is mainly
 related to interactions between OpenCL and {cpp} language features.
 
+This document describes C++ for OpenCL language:
+
+ * version 1.0 that is backward compatible with OpenCL 2.0;
+ * version 2021 that is backward compatible with OpenCL 3.0.
+
+== Version differences 
+
+The main difference  between {cpp} for OpenCL version 1.0 and version 2021
+comes from the difference between OpenCL 2.0 and OpenCL 3.0 with which they
+are respectively compatible. Support for some features of {cpp} for OpenCL 1.0
+has become optional as described in `OpenCL 3.0 s6.2.1`. Predefined feature
+macros can be used to detect if optional feature is present. 
+
+This impacts some C++ specific sematics mainly due to optionality of
+generic address space (i.e. `+__opencl_c_generic_address_space+` feature) or
+program scope variables (i.e. `__opencl_c_program_scope_global_variables`
+feature).
+
 == The {cpp} for OpenCL Programming Language 
 
-This programming language inherits features from `OpenCL C v2.0,
-s6` as well as {cpp}17. Detailed aspects of OpenCL and {cpp} are not
-described in this document as they can be found in their official
-specifications.
+This programming language inherits features from `OpenCL C 3.0, s6` as well
+as {cpp}17. Detailed aspects of OpenCL and {cpp} are not described in this
+document as they can be found in their official specifications.
  
 This section documents various language features of {cpp} for OpenCL that are
 not covered in either OpenCL or {cpp} specifications, in particular:
```

Comment:
```suggestion
not covered in neither of OpenCL C or {cpp} specifications, in particular:
```

Reply:
Agree, I think "the most correct" would be
```suggestion
not covered in neither OpenCL nor {cpp} specifications, in particular:
```